modules/videoio/src/backend_plugin.cpp

```diff
@@ -508,7 +508,26 @@ class PluginWriter : public cv::IVideoWriter
     {
         CV_Assert(plugin_api);
         CvPluginWriter writer = NULL;
-        if (plugin_api->Writer_open)
+        if (plugin_api->api_header.api_version == 1 && plugin_api->Writer_open_with_params)
+        {
+            CV_Assert(plugin_api->Writer_release);
+            CV_Assert(!filename.empty());
+            std::vector<int> vint_params;
+            for (const auto& param : params.getAll())
+            {
+                vint_params.push_back(param.key);
+                vint_params.push_back(param.value);
+            }
+            int* c_params = &vint_params[0];
+            size_t n_params = vint_params.size() / 2;
```

Comment:
I propose to extract this code as method of `VideoWriterParameters` class. E.g. return std::vector and pass it's content to C code as plain C array with size. We can potentially optimized the class behavior without plugins code modification.

Reply:
Ok I added `getIntVector()`.